src/components/PeoplePage.tsx

```diff
@@ -1,31 +1,151 @@
+import { useEffect, useMemo, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
+
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { Person } from '../types';
+import { getPeople } from '../api';
+import {
+  CenturiesOptions,
+  OrderOption,
+  SexOptions,
+  SortOptions,
+} from '../utils/data';
+import { getSearchWith } from '../utils/searchHelper';
 
 export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [currentPeople, setCurrentPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [isError, setIsError] = useState<boolean>(false);
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const sex: SexOptions = searchParams.get('sex') as SexOptions || '';
+  const centuries: CenturiesOptions[] = searchParams
+    .getAll('centuries') as CenturiesOptions[] || [];
+  const query = searchParams.get('query') || '';
+  const sort: SortOptions = searchParams.get('sort') as SortOptions || '';
+  const order: OrderOption = searchParams.get('order') as OrderOption || '';
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    getPeople()
+      .then(setPeople)
+      .catch(() => setIsError(true))
+      .finally(() => setIsLoading(false));
+  }, []);
+
+  useMemo(() => {
```

Comment:
Using useMemo without returning a value is an anti-pattern. useMemo is meant to memoize expensive computations. Consider splitting this logic into a function and call this function when needed.